src/main/kotlin/step2/SimpleExpressionParser.kt

```diff
@@ -0,0 +1,60 @@
+package step2
+
+class SimpleExpressionParser : ExpressionParser {
+
+    private val operands: MutableList<Int> = mutableListOf()
+    private val operators: MutableList<Operator> = mutableListOf()
+
+    override fun parse(expression: String) {
+        resetParser()
+
+        validateExpression(expression)
+        val tokenizedExpression = tokenizeExpression(expression)
+
+        for (i in 0..tokenizedExpression.lastIndex step 2) {
+            val token = tokenizedExpression[i]
+            val operand = token.toIntOrNull()
+            requireNotNull(operand)
```

Comment:
이렇게 null check를 해야 하는 경우 `elvis` 연산을 사용하면 조금 더 간결하게 표현할 수 있습니다!
```suggestion
        val operand = token.toIntOrNull() ?: throw IlegalArgumentException()
```